### DISCUSSION TITLE
Ability to filter by label without a match

### DISCUSSION THREAD

**Discussion by sparrowt** [fluent/fluentd#3826] [Q&A]
I tried to use `@label` in order to reduce duplication across multiple filter patterns, however I get this:
```
error_class=Fluent::ConfigError error="Missing <match> sections in <label @MYLABEL> section"
```
What I am trying to do:
```
<source>
  ...
  tag "something.long.and.complicated"
  @label @MYLABEL
</source>
<source>
  # some other source
</source>

<label @MYLABEL>
  # Filters which should only get applied to MYLABEL
  <filter **>
    ...
  </filter>
  <filter **>
    ...
  </filter>
</label>

<match **>
  # output everything to the same place
</match>
```
Unfortunately this doesn't work and I don't want to move the `<match>` inside the `<label>` block because it applies to messages from other sources/labels too.

Instead I am forced to do this:
```
<source>
  ...
  tag "something.long.and.complicated"
</source>
<source>
  # some other source
</source>

<filter something.long.and.complicated>
  ...
</filter>
<filter something.long.and.complicated>
  ...
</filter>

<match **>
  # output everything to the same place
</match>
```
Note now that "something.long.and.complicated" is repeated 3 times in my config file which isn't very [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)

Is there a reason <label> can't be used just for <filter> without the requirement to have a <match> in there too?

**Comment by adelbordbari** [comment]
same problem
i get `error_class=Fluent::ConfigError error="Missing <match> sections in <label @success> section"`

**Comment by daipom** [comment]
A label of Fluentd is an independent routing definition without `source`.
Once routed to a label, routing is usually completed within that label.
So, you must set one or more `match` to a label.

Indeed, something like a filter-only label would be useful to manage multiple combinations of `filter` sets and `match` sets.
For this, you can use [out_relabel](https://docs.fluentd.org/output/relabel) or [out_rewrite_tag_filter](https://docs.fluentd.org/output/rewrite_tag_filter) to re-route events to another label or `ROOT` (`@ROOT`).

Example:
(I have not confirmed that it works. Please try it and let me know if it does not work.)

```xml
# Source using filter-A
<source>
  @label @filter-A
  ...
</source>

# Source not using filter-A
<source>
  @label @output-A
  ...
</source>

<label @filter-A>
  <filter **>...</filter>
  <filter **>...</filter>
  <match **>
    @type relabel
    @label @output-A
  </match>
</label>

<label @output-A>
  <match **>...</match>
</label>
```
